src/QueryBuilder/QueryBuilder/QueryBuilderModal/RepeatableFields/RepeatableFields.js

```diff
@@ -39,7 +39,8 @@ export const RepeatableFields = ({ source, setSource, getParamsSource, columns }
   };
 
   const onFilter = (value, dataOptions) => {
-    return dataOptions.filter(option => new RegExp(value, 'i').test(option.label));
+    const escapedValue = value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+    return dataOptions.filter(option => new RegExp(escapedValue, 'i').test(option.label));
```

Comment:
Regexes tend to be write-once-read-never so a comment explaining what you're matching on would be super helpful. Likewise, why is backslash-dollar-ampersand the universal replacement? It's very different than what you had previously but there is no explanation in the PR description or in code comments about why this is so. 

Related, are you confident that searching for these specific special characters is correct (and you don't care about angle brackets or quotes?) or would you be better off with a simpler expression like `/[^\w\s]/` to get all non-word non-whitespace characters? Will this function and this regex do what you want with non-ASCII input, e.g. when the locale is Chinese or Arabic or any Latin script with accented characters?

Also, lint apparently wants a blank line before the return statement.